lib/cache/cache.go

```diff
@@ -1467,7 +1467,17 @@ func (c *Cache) GetRemoteCluster(clusterName string) (types.RemoteCluster, error
 		ta(cachedRemote)
 		return cachedRemote.Clone(), nil
 	}
-	return rg.presence.GetRemoteCluster(clusterName)
+	rc, err := rg.presence.GetRemoteCluster(clusterName)
+	if trace.IsNotFound(err) && rg.IsCacheRead() {
+		// release read lock early
+		rg.Release()
+		// fallback is sane because method is never used
+		// in construction of derivative caches.
+		if rc, err := c.Config.Presence.GetRemoteCluster(clusterName); err == nil {
+			return rc, nil
+		}
```

Comment:
Should we return the error when falling back instead of the original `NotFound` error?
```suggestion
		rc, err := c.Config.Presence.GetRemoteCluster(clusterName)
		return rc, trace.Wrap(err)
```

Reply:
If anything, this is a more up-to-date error, right?

Reply:
It would at least indicate via the stack trace that we made it into the fallback case

Reply:
Good question... I'm not really certain which is better, but I lean toward keeping the behavior as-is because it keeps the cache's behavior consistent in the face of network instability.  I.e. if the cache can't check for newer state (e.g. due to a network error), then it faithfully represents the last known state to the caller.  Having the cache inconsistently return either `NotFound`  some kind of networking error based on whether or not the auth server is dialable would be weird.

Regardless, if we were to change it, I would want to do that for all the methods that do this kind of fallback, and I'd rather not do that here.